### ISSUE TITLE
`from_filename` gives the wrong error when a file is found but fails to parse

### ISSUE THREAD

**Issue by sgrif** [slapresta/rust-dotenv#24]
in `.env`:

```
aslkdjf akldsjfalksdj flkajsldkf asdlkjflkase
```

code:

``` rust
extern crate dotenv;

use dotenv::DotenvError::Parsing;
use dotenv::dotenv;

fn main() {
    match dotenv() {
        Err(Parsing { line }) => println!("failed to parse {}", &line);
        Err(e) => panic!("wrong error!!! {:?}", e);
        _ => {}
    }
}
```

This code should print the parsing error and exit, but will instead panic. It appears `try_parent` assumes _all_ errors are a file not being found, when it should only assume that for `Io` errors. We should also separate `FileNotFound` from `Io`, as finding a file and having the wrong permissions is very different than the file not being there.

**Comment by slapresta** [comment]
Thanks for the bug report! I may have a go at it later.

**Comment by slapresta** [comment]
The main issue is fixed in version 0.7.1.
